app/src/main/java/ltd/evilcorp/atox/ContactRepository.kt

```diff
@@ -0,0 +1,31 @@
+package ltd.evilcorp.atox
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+class ContactRepository private constructor() {
+    companion object {
+        val instance = ContactRepository()
+    }
+
+    private val contactsByKey = HashMap<ByteArray, Contact>()
+    private val contactList = MutableLiveData<ArrayList<Contact>>()
+
+
+    fun getContact(publicKey: ByteArray): LiveData<Contact> {
```

Comment:
Wouldn't it be more useful to get contacts by friend number.

Reply:
It depends. The friend number can change if you restart Toxcore which you have to do to change connection settings. I added the ability to do so in #54 though since it was useful when doing things around the Tox thread.